app/src/main/java/swati4star/createpdf/fragment/ExceltoPdfFragment.java

```diff
@@ -0,0 +1,218 @@
+                            convertToPdf(inputName);
+                        } else {
+                            MaterialDialog.Builder builder = createOverwriteDialog(mActivity);
+                            builder.onPositive((dialog12, which) -> convertToPdf(inputName))
+                                    .onNegative((dialog1, which) -> openExcelToPdf())
+                                    .show();
+                        }
+                    }
+                })
+                .show();
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        mButtonClicked = 0;
+        switch (requestCode) {
+            case mFileSelectCode:
+                if (resultCode == RESULT_OK) {
+                    mExcelFileUri = data.getData();
+                    showSnackbar(mActivity, "Excel file selected");
+                    String fileName = mFileUtils.getFileName(mExcelFileUri);
+                    if (fileName != null) {
+                        if (fileName.endsWith(Constants.excelExtension))
+                            mFileExtension = Constants.excelExtension;
+                        else if (fileName.endsWith(Constants.excelWorkbookExtension))
+                            mFileExtension = Constants.excelWorkbookExtension;
+                        else {
+                            showSnackbar(mActivity, R.string.extension_not_supported);
+                            return;
+                        }
+                    }
+                    fileName = "Excel file: " + fileName;
+                    mTextView.setText(fileName);
+                    mTextView.setVisibility(View.VISIBLE);
+                    mCreateExcelPdf.setEnabled(true);
+                    mMorphButtonUtility.morphToSquare(mCreateExcelPdf, mMorphButtonUtility.integer());
+                }
+                break;
+        }
+        super.onActivityResult(requestCode, resultCode, data);
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (grantResults.length < 1)
+            return;
+        switch (requestCode) {
+            case PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT: {
+                if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    mPermissionGranted = true;
+                    openExcelToPdf();
+                    showSnackbar(mActivity, R.string.snackbar_permissions_given);
+                } else
+                    showSnackbar(mActivity, R.string.snackbar_insufficient_permissions);
+            }
+        }
+    }
+
+    private String trimExtension(String mExcelFileUri) {
+        String trimmed = "";
+        for (int i = 0; i < mExcelFileUri.length() - 1; i++) {
+            trimmed += mExcelFileUri.charAt(i);
+        }
+        return trimmed;
+    }
```

Comment:
Could you explain what you're doing here?

Reply:
This function was meant to trim the extension to get only the file name. But this is useless now. So will remove it. @srv-twry Please help in the functioning of the app. I have given error logs. Please help me out with that.

Reply:
add these type of functions in utils class